yamux/src/connection.rs

```diff
@@ -834,6 +857,14 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Active<T> {
         Ok(proposed)
     }
 
+    /// The ACK backlog is defined as the number of streams that have not yet been acknowledged.
+    fn ack_backlog(&mut self) -> usize {
+        self.streams
+            .values()
+            .filter(|s| !s.is_acknowledged())
+            .count()
+    }
```

Comment:
Do I understand correctly, that this counts both in- and outbound streams? The two should not be mixed, right? In other words the local node should only count the un-acknowledged **outbound** streams and depending on `self.ack_backlog() >= MAX_ACK_BACKLOG` open a new stream or wait for an existing **outbound** stream to be acknowledged.

Reply:
Yeah good catch, this needs to filter for outbound streams!